src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/ComService.kt

```diff
@@ -103,14 +103,19 @@ class ComService(
       body.getSortBy(),
     )
 
-    val resultsWithLicences: List<Pair<CaseloadResult, Licence?>> =
+    val prisonersPlusLicencesList: List<Pair<CaseloadResult, Licence?>> =
       teamCaseloadResult.map { it to getLicence(it) }
 
-    val resultsWithUnstartedLicences = findPrisonersForUnstartedRecords(resultsWithLicences)
+    val eligiblePrisonersList = findPrisonersEligibleToCreateLicence(prisonersPlusLicencesList)
 
-    val searchResults = resultsWithLicences.mapNotNull { (result, licence) ->
+    val eligiblePrisonersWithoutDraftLicences =
+      findPrisonersWithoutDraftLicences(eligiblePrisonersList, prisonersPlusLicencesList)
+
+    val eligiblePrisonersForLicenceCreation = filterPrisonersWithNonDraftLicences(prisonersPlusLicencesList)
```

Comment:
So currently: 

1.) retrieve cases for probation and join with associated licences
2.)  findPrisonersEligibleToCreateLicence(1) 
3.)  findPrisonersWithoutDraftLicences(2, 1)
4.)  filterPrisonersWithNonDraftLicences(1)
5.)  4.map(   notStartedRecord(3) VS started())  // depending on whether licence is present

Maybe some thing like this would be better?

1.) search for cases for probation for this user/team 
2.) search for all prison records for 1.)
3.) join 1.) with any associated licences and prison records
4.) 3.filter(   isEligibleNotStartedRecord() VS isEligibleStartedRecord())  // depending on whether licence is present
5.) 4.map(   createNotStartedRecord() VS createStartedRecord()) // depending on whether licence is present

Now you always need to find the associated prisoner search details, doing that upfront might be simpler before filtering.

This would allow all the filtering logic to be in one place for each type of licence, and so a bit easier to read?

The only criteria/logic this ticket should be changing is by implementing something that encapsulates the concept of `isEligibleStartedRecord()`